commit 56c888446600991803fd92d668349101ad4bf160
Author: Christoph Moench-Tegeder <cmt@burggraben.net>
Date:   Tue Feb 6 22:51:27 2024 +0100

    switch to -fvisibility flags
    
    this fixes linkage with llvm18 (which does not like the former
    approach via the #pragma in gcc_hidden.h
    
    PR: 276746
    Submitted by: dim@

diff --git build/moz.configure/toolchain.configure build/moz.configure/toolchain.configure
index d08b748db250..4696f69153f6 100644
--- build/moz.configure/toolchain.configure
+++ build/moz.configure/toolchain.configure
@@ -2186,7 +2186,7 @@ set_define("_LIBCPP_HIDE_FROM_ABI", libcxx_override_visibility.hide_from_abi)
 @depends(target, build_environment)
 def visibility_flags(target, env):
     if target.os != "WINNT":
-        if target.kernel == "Darwin":
+        if target.kernel == "Darwin" or target.kernel == "FreeBSD":
             return ("-fvisibility=hidden", "-fvisibility-inlines-hidden")
         return (
             "-I%s/system_wrappers" % os.path.join(env.dist),
